src/core/src/FabricMaterial.cpp

```diff
@@ -615,18 +629,43 @@ void FabricMaterial::initializeDefaultMaterial() {
         createConnection(srw, _baseColorTexturePath, shaderPath, FabricTokens::inputs_base_color_texture);
     }
 
-    if (imageryLayerCount == 1) {
+    if (ionImageryLayerCount == 1) {
         // Create connection from imagery layer to shader
-        const auto& imageryLayerPath = _imageryLayerPaths.front();
+        const auto& imageryLayerPath = _imageryLayerPaths[ionImageryLayerIndices[0]];
         createConnection(srw, imageryLayerPath, shaderPath, FabricTokens::inputs_imagery_layer);
-    } else if (imageryLayerCount > 1) {
+    } else if (ionImageryLayerCount > 1) {
         // Create connection from imagery layer resolver to shader
         createConnection(srw, _imageryLayerResolverPath, shaderPath, FabricTokens::inputs_imagery_layer);
 
         // Create connections from imagery layers to imagery layer resolver
-        for (uint64_t i = 0; i < imageryLayerCount; i++) {
+        int layerCounter = 0;
```

Comment:
Makes sense. But where is clang-tidy complaining for you? I'm not seeing the complaint. It would be a handy notification.
![image](https://github.com/CesiumGS/cesium-omniverse/assets/1040194/0c2aca64-ef6c-4f95-aee1-971127d478f0)